core/admin/mailu/sso/views/base.py

```diff
@@ -95,6 +94,8 @@ def proxy(target='webmail'):
         return flask.abort(500, 'Too many users in (domain=%s)' % domain)
     user = models.User(localpart=localpart, domain=domain)
     user.set_password(secrets.token_urlsafe())
+    flask.session.regenerate()
+    flask_login.login_user(user)
```

Comment:
Shouldn't this (lines 97-98) occur `after` committing the changes to the database (after `models.db.session.commit()`).
Now you are creating a session for a non-existing user.

Reply:
That's 6a4d8603